diff --git a/vendor/gateway-helm/templates/_helpers.tpl b/helm/envoy-gateway/templates/_helpers.tpl
index df9bb58..ca5a6fe 100644
--- a/vendor/gateway-helm/templates/_helpers.tpl
+++ b/helm/envoy-gateway/templates/_helpers.tpl
@@ -47,6 +47,7 @@ helm.sh/chart: {{ include "eg.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+application.giantswarm.io/team: {{ index .Chart.Annotations "io.giantswarm.application.team" | quote }}
 {{- with .Values.commonLabels }}
 {{ toYaml . }}
 {{- end }}
@@ -100,7 +101,7 @@ The name of the Envoy Gateway image.
 {{-   $imageTag := $repositoryParts._1 -}}
 {{-   printf "%s/%s:%s" $registryName $repositoryName $imageTag -}}
 {{- else -}}
-docker.io/envoyproxy/gateway:{{ .Chart.Version }}
+{{- printf "%s/giantswarm/envoyproxy-gateway:v%s" (default "gsoci.azurecr.io" .Values.global.imageRegistry) .Chart.AppVersion -}}
 {{- end -}}
 {{- end -}}
 
@@ -236,10 +237,16 @@ provider:
       {{- end }}
     shutdownManager:
       image: {{ include "eg.image" . }}
-{{- with .Values.config.envoyGateway.extensionApis }}
+{{- $extensionApis := .Values.config.envoyGateway.extensionApis | default dict -}}
+{{- if .Values.backend.enabled -}}
+{{-   $extensionApis = merge (dict "enableBackend" true) $extensionApis -}}
+{{- end -}}
+{{- with $extensionApis }}
+{{- if gt (len .) 0 }}
 extensionApis:
   {{- toYaml . | nindent 2 }}
 {{- end }}
+{{- end }}
 {{- if not .Values.topologyInjector.enabled }}
 proxyTopologyInjector:
   disabled: true
