diff --git a/vendor/gateway-helm/templates/_helpers.tpl b/helm/envoy-gateway/templates/_helpers.tpl
index df9bb58..8d4db30 100644
--- a/vendor/gateway-helm/templates/_helpers.tpl
+++ b/helm/envoy-gateway/templates/_helpers.tpl
@@ -47,6 +47,7 @@ helm.sh/chart: {{ include "eg.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+application.giantswarm.io/team: {{ index .Chart.Annotations "io.giantswarm.application.team" | quote }}
 {{- with .Values.commonLabels }}
 {{ toYaml . }}
 {{- end }}
@@ -236,10 +237,16 @@ provider:
       {{- end }}
     shutdownManager:
       image: {{ include "eg.image" . }}
-{{- with .Values.config.envoyGateway.extensionApis }}
+{{- $extensionApis := .Values.config.envoyGateway.extensionApis | default dict -}}
+{{- if .Values.backend.enabled -}}
+{{-   $extensionApis = merge (dict "enableBackend" true) $extensionApis -}}
+{{- end -}}
+{{- with $extensionApis }}
+{{- if gt (len .) 0 }}
 extensionApis:
   {{- toYaml . | nindent 2 }}
 {{- end }}
+{{- end }}
 {{- if not .Values.topologyInjector.enabled }}
 proxyTopologyInjector:
   disabled: true
