diff --git a/helm/envoy-gateway/templates/_helpers.tpl b/helm/envoy-gateway/templates/_helpers.tpl
--- a/helm/envoy-gateway/templates/_helpers.tpl
+++ b/helm/envoy-gateway/templates/_helpers.tpl
@@ -177,10 +177,17 @@ provider:
       {{- end }}
     shutdownManager:
       image: {{ include "eg.image" . }}
-{{- with .Values.config.envoyGateway.extensionApis }}
+{{- $extensionApis := .Values.config.envoyGateway.extensionApis | default dict -}}
+{{- if .Values.backend.enabled -}}
+{{-   $extensionApis = merge (dict "enableBackend" true) $extensionApis -}}
+{{- end -}}
+{{- with $extensionApis }}
+{{- if gt (len .) 0 }}
 extensionApis:
   {{- toYaml . | nindent 2 }}
 {{- end }}
+{{- end }}
 {{- if not .Values.topologyInjector.enabled }}
 proxyTopologyInjector:
   disabled: true
